### ISSUE TITLE
Performance/RegexpMatch cop behaves differently with unrelated config file

### ISSUE THREAD

**Issue by okuramasafumi** [rubocop-hq/rubocop#6739]
When `.rubocop.yml` is just like

```
Style/TrailingCommaInArguments:
  EnforcedStyleForMultiline: consistent_comma
```

the result of `Performance/RegexpMatch` cop changed, which never should change.

--------

## Expected behavior

When `.rubocop.yml` is

```
Style/TrailingCommaInArguments:
  EnforcedStyleForMultiline: consistent_comma
```

and `sample.rb` is

```
# frozen_string_literal: true

if 'foo'.match(/foo/)
  puts 'foo'
  puts 'foo'
end
```

the output from `rubocop sample.rb` should look like

```
Inspecting 1 file
C

Offenses:

sample.rb:3:4: C: Performance/RedundantMatch: Use =~ in places where the MatchData returned by #match will not be used.
if 'foo'.match(/foo/)
   ^^^^^^^^^^^^^^^^^^
sample.rb:3:4: C: Performance/RegexpMatch: Use match? instead of match when MatchData is not used.
if 'foo'.match(/foo/)
   ^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected
```

## Actual behavior

The actual output is

```
Inspecting 1 file
C

Offenses:

sample.rb:3:4: C: Performance/RedundantMatch: Use =~ in places where the MatchData returned by #match will not be used.
if 'foo'.match(/foo/)
   ^^^^^^^^^^^^^^^^^^
sample.rb:3:4: C: Performance/RegexpMatch: Use match? instead of =~ when MatchData is not used.
if 'foo'.match(/foo/)
   ^^^^^^^^^^^^^^
sample.rb:3:4: C: Performance/RegexpMatch: Use match? instead of match when MatchData is not used.
if 'foo'.match(/foo/)
   ^^^^^^^^^^^^^^^^^^

1 file inspected, 3 offenses detected
```

## Steps to reproduce the problem

Run `rubocop sample.rb` with the given content above. 

## RuboCop version

`0.63.1 (using Parser 2.6.0.0, running on ruby 2.6.0 x86_64-darwin15)`

**Comment by Drenmi** [comment]
I was unable to reproduce this on current `master` with the configuration above. Using 2.6 as the target Ruby version, I get the expected output regardless of configuring `TrailingCommaInArguments`. Maybe something is missing from the reproduction steps?

**Comment by okuramasafumi** [comment]
@Drenmi thanks for the response. I added `TargetRubyVersion: 2.6` to the config file and confirmed there is no problem. I didn't have an `TargetRubyVersion` entry as an reproduction step.

Here's a weird thing: I had a typo in the config file (`AllCops` -> `All`) and the problem has gone. I tried to add completely nonsense entry to the top of the config file (say, `Foo:\n Bar: true` or something like that) and it also solves the problem.
Is it related to YAML parsing?
